src/calculate.js

```diff
@@ -34,10 +34,16 @@ function compareMembersByDayLevel(day, level) {
 }
 
 // note: day is 0-based here
-function compareRankingByDay(day) {
+function compareRankingByDay(day, fallback) {
     return function (a, b) {
         if (a.scores[day] === b.scores[day]) {
-            return +a.id - +b.id;
+            if (fallback) {
+                return +a.id - +b.id;
+            }
+            if (day <= 0) {
+                return compareRankingByDay(day + 1, true)(a, b);
+            }
```

Comment:
Prevent endless loop between day 0 & 1 by passing `fallback` parameter to only look forward (next day) once. Backwards (previous day) will be looked up recursively until day 0.